src/interface.jl

```diff
@@ -0,0 +1,655 @@
+logabsdetjac(::Identity, y::AbstractArray{T}) where T <: Real = zero(T)
+
+const IdentityBijector = Identity()
+
+###############################
+# Example: Logit and Logistic #
+###############################
+using StatsFuns: logit, logistic
+
+struct Logit{T<:Real} <: Bijector
+    a::T
+    b::T
+end
+
+(b::Logit)(x) = @. logit((x - b.a) / (b.b - b.a))
+(ib::Inversed{<: Logit{<: Real}})(y) = @. (ib.orig.b - ib.orig.a) * logistic(y) + ib.orig.a
+
+logabsdetjac(b::Logit{<:Real}, x) = @. log((x - b.a) * (b.b - x) / (b.b - b.a))
+
+#############
+# Exp & Log #
+#############
+
+struct Exp <: Bijector end
+struct Log <: Bijector end
+const exp_b = Exp()
+const log_b = Log()
+
+(b::Log)(x) = @. log(x)
+(b::Exp)(y) = @. exp(y)
+
+inv(b::Log) = exp_b
+inv(b::Exp) = log_b
+
+logabsdetjac(b::Log, x) = log(x)
+logabsdetjac(b::Exp, y) = - y
+
+#################
+# Shift & Scale #
+#################
+struct Shift{T} <: Bijector
+    a::T
+end
+
+(b::Shift)(x) = b.a + x
+(b::Shift{<: Real})(x::AbstractArray) = b.a .+ x
+(b::Shift{<: AbstractVector})(x::AbstractMatrix) = b.a .+ x
+
+inv(b::Shift) = Shift(-b.a)
+logabsdetjac(b::Shift, x) = zero(eltype(x))
+# FIXME: ambiguous whether or not this is actually a batch or whatever
+logabsdetjac(b::Shift{<: Real}, x::AbstractMatrix) = zeros(eltype(x), size(x, 2))
+logabsdetjac(b::Shift{<: AbstractVector}, x::AbstractMatrix) = zeros(eltype(x), size(x, 2))
+
+struct Scale{T} <: Bijector
+    a::T
+end
+
+(b::Scale)(x) = b.a * x
+(b::Scale{<: Real})(x::AbstractArray) = b.a .* x
+(b::Scale{<: AbstractVector{<: Real}})(x::AbstractMatrix{<: Real}) = x * b.a
+
+inv(b::Scale) = Scale(inv(b.a))
+inv(b::Scale{<: AbstractVector}) = Scale(inv.(b.a))
+
+# TODO: should this be implemented for batch-computation?
+# There's an ambiguity issue
+#      logabsdetjac(b::Scale{<: AbstractVector}, x::AbstractMatrix)
+# Is this a batch or is it simply a matrix we want to scale differently
+# in each component?
+logabsdetjac(b::Scale, x) = log(abs(b.a))
+
+####################
+# Simplex bijector #
+####################
+struct SimplexBijector{T} <: Bijector where T end
+
+const simplex_b = SimplexBijector{Val{false}}()
+const simplex_b_proj = SimplexBijector{Val{true}}()
+
+# the following implementations are basically just copy-paste from `invlink` and
+# `link` for `SimplexDistributions` but dropping the dependence on the `Distribution`.
+function _clamp(x::T, b::SimplexBijector) where T
+    bounds = (zero(T), one(T))
+    clamped_x = clamp(x, bounds...)
+    DEBUG && @debug "x = $x, bounds = $bounds, clamped_x = $clamped_x"
+    return clamped_x
+end
+
+function (b::SimplexBijector{Val{proj}})(x::AbstractVector{T}) where {T, proj}
+    y, K = similar(x), length(x)
+
+    ϵ = _eps(T)
+    sum_tmp = zero(T)
+    @inbounds z = x[1] * (one(T) - 2ϵ) + ϵ # z ∈ [ϵ, 1-ϵ]
+    @inbounds y[1] = StatsFuns.logit(z) + log(T(K - 1))
+    @inbounds @simd for k in 2:(K - 1)
+        sum_tmp += x[k - 1]
+        # z ∈ [ϵ, 1-ϵ]
```

Comment:
What are these lines?

Reply:
You mean the comments or the entire thing?
This is just copy-pasted from the `logpdf_with_trans`, `link` and `invlink` for `SimplexDistribution`. So those questions are probably better directed at the person who implemented this, but if you're referring to the comment I think it's just stating that we've done a "small" rescaling to ensure that `z` is _not_ on the boundary. I just assumed that we do this because it ensures that the transformation is invertible or differentiable. No closed interval can be isomorphic to the real-line, so by instead considering `[ϵ, 1-ϵ]` we're instead considering the "largest"_open_ interval inside of `[0 ,1]` which indeed _is_ isomorphic to the real line, i.e. we have a bijection:)

If you're asking _why_ we have this comment, I don't know but I'm gladwe do! Personally I find it helpful.

Reply:
Yeah, I think either @mohamed82008 or I added these comments at some point (probably @mohamed82008 ). Probably better to keep them around for exactly this type of situation.

Reply:
SGTM!